### PR TITLE
[12.0][IMP] mail_tracking: ignore catchall

### PR DESCRIPTION
**Pull request by Tardo**
- Improve tests
- Ignore catchall from suggested recipients

cc @tecnativa @rafaelbn
**Needs to be ported to 13.0**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please check Travis

**Comment by Tardo** [comment]
Changes done

**Comment by Tardo** [comment]
Please review @rafaelbn

File: mail_tracking/models/mail_alias.py [hunk 1]
```diff
@@ -10,9 +10,15 @@ class MailAlias(models.Model):
     @api.model
     @tools.ormcache()
     def get_aliases(self):
-        return set(x['display_name'] for x in self.search_read([
+        aliases = set(x['display_name'] for x in self.search_read([
             ('alias_name', '!=', False),
         ], ['display_name']))
+        IrConfigParamObj = self.env["ir.config_parameter"].sudo()
+        catchall = "%s@%s" % (
```
Comment (pedrobaeza) [inline mail_tracking/models/mail_alias.py:17]: The rest of the aliases don't contain the domain. Why this one does?

**Review by rafaelbn — APPROVED** [review]
:+1: Thanks

File: mail_tracking/models/mail_alias.py (hunk 1 above)
Comment (Tardo) [reply mail_tracking/models/mail_alias.py:17]: With mail.alias records we are using "display_name" that includes the catchall domain.

**Review by pedrobaeza — COMMENTED** [review]
As it's not usual to edit/create/unlink ir.config.parameter, just clear cache always in these cases for avoiding excessive code.

File: mail_tracking/models/mail_alias.py [hunk 2]
```diff
@@ -10,9 +10,15 @@ class MailAlias(models.Model):
...
+            IrConfigParamObj.get_param("mail.catchall.alias"),
+            IrConfigParamObj.get_param("mail.catchall.domain"))
+        aliases.update({catchall})
```
Comment (pedrobaeza) [inline mail_tracking/models/mail_alias.py]: Isn't simpler to add the element?:
```suggestion
        aliases.add(catchall)
```

**Comment by Tardo** [comment]
Changes done

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Please move it to v13:

/ocabot merge patch

**Pull request merged**